doc/src/devdocs/types.md

```diff
@@ -455,129 +335,34 @@ breakpoints in other functions.
-
-`jltypes.c` contains three closely related collections of functions for testing how types `a`
-and `b` are ordered:
-
-  * The `subtype` functions implement `a <: b`. Among other uses, they serve in matching function
-    arguments against method signatures in the function cache.
-  * The `type_morespecific` functions are used for imposing a partial order on functions in method
-    tables (from most-to-least specific). Note that `jl_type_morespecific(a,b,0)` really means "is
-    `a` at least as specific as `b`?" and not "is `a` strictly more specific than `b`?"
-  * The `type_match` functions are similar to `type_morespecific`, but additionally accept (and employ)
-    an environment to constrain typevars. The related `type_match_morespecific` functions call `type_match`
-    with an argument `morespecific=1`
-
-All three of these take an argument, `invariant`, which is set to 1 when comparing type parameters
-and otherwise is 0.
+The `type_morespecific` functions are used for imposing a partial order on functions in method
+tables (from most-to-least specific). Note that `jl_type_morespecific(a,b)` really means "is `a`
+at least as specific as `b`?" and not "is `a` strictly more specific than `b`?"
 
-The rules for these are somewhat different. `subtype` is sensitive to the number arguments, but
-`type_morespecific` may not be. In particular, `Tuple{Int,AbstractFloat}` is more specific than
-`Tuple{Integer}`, even though it is not a subtype.  (Of `Tuple{Int,AbstractFloat}` and `Tuple{Integer,Float64}`,
-neither is more specific than the other.)  Likewise, `Tuple{Int,Vararg{Int}}` is not a subtype
-of `Tuple{Integer}`, but it is considered more specific. However, `morespecific` does get a bonus
-for length: in particular, `Tuple{Int,Int}` is more specific than `Tuple{Int,Vararg{Int}}`.
+If `a` is a subtype of `b`, then it is automatically considered more specific.
+From there, `type_morespecific` employs some less formal rules.
+For example, `subtype` is sensitive to the number arguments, but `type_morespecific` may not be.
+In particular, `Tuple{Int,AbstractFloat}` is more specific than `Tuple{Integer}`, even though it is
+not a subtype.  (Of `Tuple{Int,AbstractFloat}` and `Tuple{Integer,Float64}`, neither is more specific
+than the other.)  Likewise, `Tuple{Int,Vararg{Int}}` is not a subtype of `Tuple{Integer}`, but it is
+considered more specific. However, `morespecific` does get a bonus for length: in particular,
+`Tuple{Int,Int}` is more specific than `Tuple{Int,Vararg{Int}}`.
 
 If you're debugging how methods get sorted, it can be convenient to define the function:
 
 ```julia
-args_morespecific(a, b) = ccall(:jl_args_morespecific, Cint, (Any,Any), a, b)
+type_morespecific(a, b) = ccall(:jl_type_morespecific, Cint, (Any,Any), a, b)
 ```
 
-which allows you to test whether arg-tuple `a` is more specific than arg-tuple `b`.
+which allows you to test whether argument tuple type `a` is more specific than tuple type `b`.
```

Comment:
For consistency, perhaps either "whether tuple type `a` is more specific than tuple type `b`" or "whether argument tuple type `a` is more specific than argument tuple type `b`"?